### ISSUE TITLE
Problem occurred when using dynamic bicycle model

### ISSUE THREAD

**Issue by YuqiaoBai** [eleurent/highway-env#411]
Hi @eleurent, thanks for your work.
Recently I want to create vehicles with bicycle model so I just simply change the vehicles type to "BicycleVehicle" when creating vehicles, but after that, the vehicles' behavior turned weird.

https://user-images.githubusercontent.com/16231913/216801403-f0b9e32e-583d-4dc0-9dcb-9a7cb5c764d2.mp4

**Comment by YuqiaoBai** [comment]
As shown in the video, the green vehicle turns bit crazy. I'd love to know what's causing this. Look forward to your reply

**Comment by eleurent** [comment]
This is probably because the bicycle model is a higher-order, more accurate model, so it requires higher simulation frequency (smaller integration step) otherwise it can be unstable. Do you still have issues if you increase the simulation frequency to, say, 15?

**Comment by YuqiaoBai** [comment]
> This is probably because the bicycle model is a higher-order, more accurate model, so it requires higher simulation frequency (smaller integration step) otherwise it can be unstable. Do you still have issues if you increase the simulation frequency to, say, 15?

the default simulation frequency is 15, and this time I tried to run the case with higher frequency however the problem is still there.

**Comment by YuqiaoBai** [comment]
I think I can't simply use the dynamic bicycle model class to create vehicles in envs right? There should be a controller like IDM?

**Comment by eleurent** [comment]
> I think I can't simply use the dynamic bicycle model class to create vehicles in envs right? There should be a controller like IDM?

No this should be fine, in principle your agent should be able to implement the same function (or a similar one) as a low-level controller like IDM.
And this is supported natively by configuring the action type as continuous:
```
config = {
  'action': {'type': 'ContinuousAction', 'dynamical': False},
  'simulation_frequency': 15,
  'policy_frequency': 15,
}
env = gym.make("highway-v0", config=config)
```
Using the config `dynamical=False` creates a simple Vehicle, which uses the Kinematics Bicycle Model, and should be more stable.
However, setting the config `dynamical=True` uses a more accurate dynamical Bicycle Model which models forces like tire friction, etc. However, this model is much less stable at a low simulation frequency.

> the default simulation frequency is 15, and this time I tried to run the case with higher frequency however the problem is still there.

You are right, I tried it myself and was able to reproduce these instabilities also with a higher sim frequency (especially at low speed). To make it more stable, I replaced the simple forward Euler integration scheme by an RK4. Let me know if it solves your issues, but I find it much better now.
